browser/ipfs/ipfs_host_resolver.cc

```diff
@@ -0,0 +1,94 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <utility>
+#include <vector>
+
+#include "brave/browser/ipfs/ipfs_host_resolver.h"
+#include "mojo/public/cpp/bindings/receiver.h"
+#include "net/base/host_port_pair.h"
+#include "net/dns/public/dns_protocol.h"
+#include "services/network/network_context.h"
+
+namespace {
+const uint8_t kGooglePublicDns1[] = {8, 8, 8, 8};
+const uint8_t kGooglePublicDns2[] = {8, 8, 4, 4};
```

Comment:
are we allowed to use this without users agreeing to the google DNS terms of service? https://developers.google.com/speed/public-dns/terms

Reply:
it is already used in chromium here https://source.chromium.org/chromium/chromium/src/+/master:chrome/browser/net/dns_probe_service_factory.cc;l=292;drc=6531b4f2fda5515c3315e50d1cfb709e607d53e6;bpv=1;bpt=1, 
I tried to use default DNS it works only for A/AAAA records and reports errors for attempts to read TXT

Reply:
Removed this code and pushed new one same as in dns_probe to meet actual browser prefs around DoH and secure dns, thanks